src/agni/server/IPChannelPair.java

```diff
@@ -0,0 +1,10 @@
+package agni.server;
+
+public class IPChannelPair<I,C> {
```

Comment:
Since you named the class `IPChannelPair` it doesn't quite make sense to make it a template since the types are already implied from the class name.